cadcdata/cadcdata/storageinv.py

```diff
@@ -545,14 +547,22 @@ def cadcremove(self, id):
                 logger.info('{} removed in {} ms'.format(id, duration))
                 return
             except Exception as e:
-                # try a different URL
                 logger.debug('WARN: Cannot remove data from {}. Exception: {}'.
                              format(url, e))
+                error_msg += str(e) + '\n'
                 if urls:
+                    # try a different URL
                     logger.debug('Try the next URL')
                 continue
-        raise exceptions.HttpException(
-            'Unable to remove data from any of the available URLs')
+        # no successful DELETE so far. Double check the existence of file
+        # in global
+        try:
+            self.cadcinfo(id=id)
+        except exceptions.NotFoundException:
+            # success
+            logger.debug('{} not in global anymore. File removed')
+            raise exceptions.NotFoundException(id)
```

Comment:
I do not understand why the comment is "success", but the NotFoundException is raised. Doesn't NotFoundException mean the user got the behaviour they wanted?

Reply:
Wrong comment. Removed